pkg/rc/farm.go

```diff
@@ -72,35 +81,14 @@ START_LOOP:
 	for {
 		select {
 		case <-quit:
-			rcf.logger.NoFields().Infoln("Halt requested, releasing replication controllers")
+			rcf.logger.NoFields().Infoln("Session expired, releasing replication controllers")
+			rcf.lock = nil
 			rcf.releaseChildren()
 			return
-		case session := <-rcf.sessions:
```

Comment:
Yes, and that's the main point of this change. I think it's easier to express the intended behavior as a single main loop that dies whenever its one session is lost, compared to being a long-running loop that handles sessions coming and going itself.